### PR TITLE
Performance audit

### PR DESCRIPTION
**Pull request by smunoz1988**
Hello, dear reviewer!

Team: Julio Quezada / Santiago Muñoz

For this milestone, We made these changes:

Preparation for the project:

- Made sure that our database is set up with the schema and data from our previous projects.
- Run the following query to add an extra column to the `owners ` table: 

ALTER TABLE owners ADD COLUMN email VARCHAR(120);

- Run the following statements to add data to our database:
`INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;`

`insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';`

- Check that by running explain analyze `SELECT COUNT(*) FROM visits where animal_id = 4:` - If you get Execution time: X ms and X >= 1000: that should be enough, you can continue to the project requirements.

image

Requested queries to improve:

CASE 1

    SELECT COUNT(*) FROM visits where animal_id = 4;

CASE 2

    SELECT * FROM visits where vet_id = 2;

CASE 3

    SELECT * FROM owners where email = 'owner_18327@mail.com';

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -144,3 +144,30 @@ SELECT COUNT(*) FROM visits v JOIN animals a ON v.animal_id = a.id JOIN vets ve
...
+
+CREATE INDEX animal_id_index ON visits (animal_id);
+
+/* Case 2 */
+
+SELECT * FROM visits where vet_id = 2;
+
+/* Solution to improve performance */
+
+CREATE INDEX vet_clinic ON visits(vet_id, id, animal_id, date_of_visit);
```
Comment (Owusu-Desmond) [inline queries.sql:165]: - [x] Please your second screenshot using the `EXPLAIN ANALYZE` for  `SELECT * FROM visits where vet_id = 2;` that shows how much time the query takes AFTER the improvements introduced must have the execution time significantly lower than 1 sec (1000 ms) to show there is much improvement. :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @smunoz1988 and @Alejandroq12 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Alejandroq12** [comment]
Hello, dear reviewer.

Here you have the  problem related to the second query fixed:
CASE 2

    SELECT * FROM visits where vet_id = 2;



Have a very good day.

File: queries.sql (hunk 1 above)
Comment (Alejandroq12) [reply queries.sql:165]: Thank you very much for your feedback.

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @smunoz1988     👋

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good job with the use of GitHub features✔️ 
- Kudos on the use of SQL queries ✔️ 
- Good job with the optimizations✔️ 

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Alejandroq12** [comment]
Thank you very much for your time.

**Pull request merged**